sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs

```diff
@@ -33,4 +42,47 @@ public ModelReaderWriterOptions (string format)
     /// Gets the format to read and write the model.
     /// </summary>
     public string Format { get; }
+
+    /// <summary>
+    /// .
+    /// </summary>
+    public void AddProxy<T>(Type type, IPersistableModel<T> proxy)
+    {
+        if (_proxies is null)
+        {
+            _proxies = new Dictionary<Type, object>();
+        }
+
+        //multiple of same type?
+        _proxies.Add(type, proxy);
+    }
+
+    /// <summary>
+    /// .
+    /// </summary>
+    public object? WriteContext { get; private set; }
+
+    /// <summary>
+    /// .
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="model"></param>
+    /// <returns></returns>
+    public IPersistableModel<T> GetPersistableInterface<T>(IPersistableModel<T> model)
+    {
+        WriteContext = model;
+        return Proxies.TryGetValue(typeof(T), out var proxy) && proxy is IPersistableModel<T> proxyAsT ? proxyAsT : model;
```

Comment:
this will create the dictionary, even if there are no proxies. Can we optimize it out?